lang/image.bzl

```diff
@@ -250,7 +271,7 @@ image = struct(
         # the runfiles dir.
         "binary": attr.label(
             executable = True,
-            cfg = "target",
+            cfg = image_transition,
```

Comment:
what would users do if they're using `container_image` rather than one of the convenience lang-specific rules? do they need to make their own transition, or just refer to this one?

Reply:
The `binary` and `dep` attributes are not defined on the `container_image` rule and my initial thoughts were that it might not be necessary to include this anywhere else.

I thought about it for a bit and I think you raise a good point. It might be necessary to include the transition for all label attributes to ensure everything is built for the correct target.

A quick look at which attributes may need to have an attached transition:
- [container_layer.files](https://github.com/bazelbuild/rules_docker/blob/61e2df80a0f92c2c9e96c17390b1ecc64af65e40/container/layer.bzl#L342)
- [container_layer.tars](https://github.com/bazelbuild/rules_docker/blob/61e2df80a0f92c2c9e96c17390b1ecc64af65e40/container/layer.bzl#L369)
- [container_image.base](https://github.com/bazelbuild/rules_docker/blob/61e2df80a0f92c2c9e96c17390b1ecc64af65e40/container/image.bzl#L585)
- [container_image.label_files](https://github.com/bazelbuild/rules_docker/blob/61e2df80a0f92c2c9e96c17390b1ecc64af65e40/container/image.bzl#L665)
- [container_image.launcher](https://github.com/bazelbuild/rules_docker/blob/61e2df80a0f92c2c9e96c17390b1ecc64af65e40/container/image.bzl#L686)
- [container_image.layers](https://github.com/bazelbuild/rules_docker/blob/61e2df80a0f92c2c9e96c17390b1ecc64af65e40/container/image.bzl#L700)
- [lang/image.base](https://github.com/bazelbuild/rules_docker/blob/61e2df80a0f92c2c9e96c17390b1ecc64af65e40/lang/image.bzl#L245)
- [lang/image.binary](https://github.com/bazelbuild/rules_docker/blob/61e2df80a0f92c2c9e96c17390b1ecc64af65e40/lang/image.bzl#L251) (done)
- [lang/image.data](https://github.com/bazelbuild/rules_docker/blob/61e2df80a0f92c2c9e96c17390b1ecc64af65e40/lang/image.bzl#L258)
- [lang/image.dep](https://github.com/bazelbuild/rules_docker/blob/61e2df80a0f92c2c9e96c17390b1ecc64af65e40/lang/image.bzl#L266) (done)


Another thought: I wonder if transitions affect `select`? If so, would this mean basically every attribute would need to be transitioned? Assuming this is the case, it might make sense to use an [incoming edge transition](https://docs.bazel.build/versions/main/skylark/config.html#incoming-edge-transitions) instead.

I'll test that incoming edge transition idea and get back to you.

Reply:
Just had a quick play with incoming edge transitions and I don't think they're what we're looking for.

Reply:
Incoming edge transitions seem to work as expected. I can update this PR to use them instead as that will then transition the entire target, rather than just individual attributes. Attributes with `cfg = "host"` will be respected.

Reply:
Done.

All rules which inherit [container/image](https://github.com/bazelbuild/rules_docker/blob/61e2df80a0f92c2c9e96c17390b1ecc64af65e40/container/image.bzl#L802) now transition to the correct target platform. Dependencies of these rules also implicitly transition (i.e [container_layer](https://github.com/bazelbuild/rules_docker/blob/61e2df80a0f92c2c9e96c17390b1ecc64af65e40/docs/container.md#container_layer)).